### PR TITLE
Add `ext-mbstring` to list of suggested dependencies

### LINKED ISSUE #138: Call to undefined function WP_CLI\I18n\mb_ereg() in phar
wp i18n make-pot . languages/my-plugin.pot
Theme stylesheet detected.
PHP Fatal error:  Uncaught Error: Call to undefined function WP_CLI\I18n\mb_ereg() in phar:///usr/local/bin/wp/vendor/wp-cli/i18n-command/src/IterableCodeExtractor.php:145
Stack trace:
#0 phar:///usr/local/bin/wp/vendor/wp-cli/i18n-command/src/IterableCodeExtractor.php(173): WP_CLI\I18n\PhpCodeExtractor::isExcluded(Object(SplFileInfo), Array)
#1 [internal function]: WP_CLI\I18n\PhpCodeExtractor::WP_CLI\I18n\{closure}(Object(SplFileInfo), '/var/www/html/g...', Object(RecursiveDirectoryIterator))
#2 [internal function]: CallbackFilterIterator->accept()
#3 phar:///usr/local/bin/wp/vendor/wp-cli/i18n-command/src/IterableCodeExtractor.php(191): FilterIterator->rewind()
#4 phar:///usr/local/bin/wp/vendor/wp-cli/i18n-command/src/IterableCodeExtractor.php(83): WP_CLI\I18n\PhpCodeExtractor::getFilesFromDirectory('/var/www/html/g...', Array, Array, Array)
#5 phar:///usr/local/bin/wp/vendor/wp-cli/i18n-command/src/MakePotCommand.php(529): WP_CLI\I18n\PhpCodeExtractor::fromDirectory('/var/www/html/g...', Object(Gettext\Translations), Arra in phar:///usr/local/bin/wp/vendor/wp-cli/i18n-command/src/IterableCodeExtractor.php on line 145

### PR DESCRIPTION
**Pull request by swissspidy**
`IterableCodeExtractor` uses `mb_ereg`.

Fixes #138

### CODE REVIEW

**Comment by schlessera** [comment]
I cannot merge this change, as it would effectively force this requirement for the entire bundled WP-CLI.

Instead of adding a hard requirement to Composer, it should be a suggested optional dependency, and the command should have a `before_invoke` action that checks for the presence of the extension and exits with a `WP_CLI::error()` if it is not present.

**Comment by swissspidy** [comment]
> Instead of adding a hard requirement to Composer, it should be a suggested optional dependency, and the command should have a `before_invoke` action that checks for the presence of the extension and exits with a `WP_CLI::error()` if it is not present.

Makes sense. Just made the necessary changes 👍

File: composer.json [hunk 1]
```diff
@@ -19,6 +19,9 @@
         "wp-cli/scaffold-command": "^1.2 || ^2",
         "wp-cli/wp-cli-tests": "^3.0.11"
     },
+    "suggest": {
+        "ext-mbstring": "Used for calculation include/exclude matches in code extraction"
```
Comment (schlessera) [inline composer.json:23]:
```suggestion
        "ext-mbstring": "Used for calculating include/exclude matches in code extraction"
```

**Review by schlessera — CHANGES_REQUESTED** [review]

**Review by schlessera — APPROVED** [review]

File: composer.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
         "wp-cli/wp-cli-tests": "^3.0.11"
     },
     "suggest": {
-        "ext-mbstring": "Used for calculation include/exclude matches in code extraction"
+        "ext-mbstring": "Used for calculating include/exclude matches in code extraction"
```
Comment (swissspidy) [inline composer.json:23]: Thanks!

**Pull request merged**